### PR TITLE
Core/Arena/Custom: Dynamic arena matchmaking rating system

### PR DESCRIPTION
**Pull request by pjasicek**
This is custom (non-blizzlike) code which solves problems of many high-rated arena teams who wait too long to be matched against opponents within their matchmaking rating range or they even need to wait for discard time ro run off and they are matched against random low-mmr opponents.

This code periodically increases team's matchmaking range (configurable in config file).

This system is disabled by default in config.

### CODE REVIEW

**Comment by ShinDarth** [comment]
Thank you for this PR, please squash your commits: https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/

**Comment by Demonid** [comment]
Nice, let's hope this gets merged.

**Comment by Pro100Hens** [comment]
http://pastebin.com/seMTTnJc
1176 line: http://prntscr.com/7cxjoe
debug crash.

File: src/server/worldserver/worldserver.conf.dist [hunk 1]
```diff
@@ -3361,3 +3361,32 @@ PacketSpoof.BanDuration = 86400
 
 #
 ###################################################################################################
+
+###################################################################################################
+# ARENA DYNAMIC MATCHMAKING SETTINGS
+#
+#    ArenaDynamicMatchmakingSystem.enable
```
Comment (Carbenium) [inline src/server/worldserver/worldserver.conf.dist]: ArenaDynamicMatchmakingSystem.Enable please. Futher down the same.

Comment (tkrokli) [inline src/server/worldserver/worldserver.conf.dist]: enable => Enable (capital first letter in keywords)
update => Update (capital first letter in keywords)

**Comment by Frytiks** [comment]
Have crash. Using only command to bring players table for queue arena. (Sending packet)

http://pastebin.com/7Yb57XPk

**Comment by Aokromes** [comment]
This branch has conflicts that must be resolved
 Use the command line to resolve conflicts before continuing.

+ the crash reported.

**Comment by tkrokli** [comment]
@pjasicek : you need to update your source branch `pjasicek:3-3-5_dynamic_matchmaking` to get it up to date with `TrinityCore:3.3.5` - currently, your PR branch is 656 commits behind. This causes the PR to be out of sync with the current TC source, making the PR unreliable in comparing it to TC source.

**Comment by jackpoz** [comment]
This PR has unresolved crashes, reopen it once those crashes will be fixed.

**Comment by Voidx** [comment]
some news?

**Comment by tkrokli** [comment]
No, there won't be any news unless @pjasicek resurrects his github activity. Not been active here on github since he opened PR #14785 on 28 May 2015 (which was closed 18 days ago, 5. January 2016).

**Pull request closed without merging**